src/prefect/blocks/abstract.py

```diff
@@ -0,0 +1,69 @@
+from abc import ABC, abstractmethod
+
+from prefect import get_run_logger
+from prefect.blocks.core import Block
+from prefect.exceptions import MissingContextError
+from prefect.logging.loggers import get_logger
+
+
+class JobRun(ABC):  # not a block
+    """
+    Represents a job run in an external system that can wait
+    for the job run's completion and fetch its results.
+    """
+
+    @property
+    def logger(self):
+        """
+        Returns a logger based on whether the JobBlock
+        is called from within a flow or task run context.
+        If a run context is present, the logger property returns a run logger.
+        Else, it returns a default logger labeled with the class's name.
+        """
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    async def wait_for_completion(self, *args, **kwargs):
+        """
+        Wait for the job run to complete and check its status
+        periodically using the interval_seconds attribute
+        until the job run is complete or the timeout_seconds
+        time limit is reached.
+        """
+
+    @abstractmethod
+    async def fetch_results(self, *args, **kwargs):
+        """
+        Retrieve the results of the job run and return them.
+        This will error if the job run is still running.
+        """
+
+
+class JobBlock(Block, ABC):
+    """
+    The JobBlock class represents a resource that can trigger a job
+    in an external service, such as Airbyte, Databricks, or dbt Cloud.
+    """
+
+    @property
+    def logger(self):
+        """
+        Returns a logger based on whether the JobBlock
+        is called from within a flow or task run context.
+        If a run context is present, the logger property returns a run logger.
+        Else, it returns a default logger labeled with the class's name.
+        """
+        try:
+            return get_run_logger()
+        except MissingContextError:
+            return get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    async def trigger(self, *args, **kwargs) -> JobRun:
```

Comment:
Similar comment to the methods on `JobRun`
```suggestion
    async def trigger(self) -> JobRun:
```